packages/engine/Specs/Scene/Model/RuntimeModelInstancingPipelineStageSpec.js

```diff
@@ -0,0 +1,289 @@
+import {
+  Cartesian3,
+  combine,
+  GltfLoader,
+  RuntimeModelInstancingPipelineStage,
+  ModelInstancesUpdateStage,
+  Math as CesiumMath,
+  Buffer,
+  BufferUsage,
+  ModelStatistics,
+  Resource,
+  ResourceCache,
+  SceneMode,
+  ShaderBuilder,
+  ModelInstance,
+  Transforms,
+  HeadingPitchRoll,
+  Ellipsoid,
+} from "../../../index.js";
+import createScene from "../../../../../Specs/createScene.js";
+import waitForLoaderProcess from "../../../../../Specs/waitForLoaderProcess.js";
+import ShaderBuilderTester from "../../../../../Specs/ShaderBuilderTester.js";
+
+describe(
+  "Scene/Model/RuntimeModelInstancingPipelineStage",
+  function () {
+    const sampleGltfUrl =
+      "./Data/Models/glTF-2.0/CesiumMilkTruck/glTF/CesiumMilkTruck.glb";
+
+    let scene;
+    let scene2D;
+    const gltfLoaders = [];
+
+    beforeAll(function () {
+      scene = createScene();
+      scene.renderForSpecs();
+
+      scene2D = createScene();
+      scene2D.renderForSpecs();
+    });
+
+    afterAll(function () {
+      scene = createScene();
+      scene2D.destroyForSpecs();
+    });
+
+    afterEach(function () {
+      const gltfLoadersLength = gltfLoaders.length;
+      for (let i = 0; i < gltfLoadersLength; ++i) {
+        const gltfLoader = gltfLoaders[i];
+        if (!gltfLoader.isDestroyed()) {
+          gltfLoader.destroy();
+        }
+      }
+      gltfLoaders.length = 0;
+      ResourceCache.clearForSpecs();
+    });
+
+    const samplePosition1 = new Cartesian3(10, 10, 10);
+    const samplePosition2 = new Cartesian3(20, 20, 20);
+
+    const headingPositionRoll = new HeadingPitchRoll();
+    const fixedFrameTransform = Transforms.localFrameToFixedFrameGenerator(
+      "north",
+      "west",
+    );
+    const instanceModelMatrix1 = new Transforms.headingPitchRollToFixedFrame(
+      samplePosition1,
+      headingPositionRoll,
+      Ellipsoid.WGS84,
+      fixedFrameTransform,
+    );
+
+    const instanceModelMatrix2 = new Transforms.headingPitchRollToFixedFrame(
```

Comment:
Do either of these test scale transforms?

Reply:
add logic to also test a scale transform